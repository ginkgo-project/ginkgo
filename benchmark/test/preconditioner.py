#!/usr/bin/env python3
import test_framework

# check that all input modes work:
# parameter
test_framework.compare_output(
    ["-input", '[{"size": 100, "stencil": "7pt"}]'],
    expected_stdout="preconditioner.simple.stdout",
    expected_stderr="preconditioner.simple.stderr",
)

# stdin
test_framework.compare_output(
    [],
    expected_stdout="preconditioner.simple.stdout",
    expected_stderr="preconditioner.simple.stderr",
    stdin='[{"size": 100, "stencil": "7pt"}]',
)

# input file
test_framework.compare_output(
    ["-input", str(test_framework.sourcepath / "input.mtx.json")],
    expected_stdout="preconditioner.simple.stdout",
    expected_stderr="preconditioner.simple.stderr",
)

# input matrix file
test_framework.compare_output(
    ["-input_matrix", str(test_framework.matrixpath)],
    expected_stdout="preconditioner.matrix.stdout",
    expected_stderr="preconditioner.matrix.stderr",
)

# set preconditioner works
test_framework.compare_output(
    [
        "-preconditioners",
        "jacobi",
        "-jacobi_max_block_size",
        "32",
        "-jacobi_storage",
        "0,0",
        "-input",
        '[{"size": 100, "stencil": "7pt"}]'],
    expected_stdout="preconditioner.precond.stdout",
    expected_stderr="preconditioner.precond.stderr",
)

# profiler annotations
test_framework.compare_output(
    [
        "-input",
        '[{"size": 100, "stencil": "7pt"}]',
        "-profile",
        "-profiler_hook",
        "debug",
    ],
    expected_stdout="preconditioner.profile.stdout",
    expected_stderr="preconditioner.profile.stderr",
)

# stdin
test_framework.compare_output(
    ["-reorder", "amd"],
    expected_stdout="preconditioner.reordered.stdout",
    expected_stderr="preconditioner.reordered.stderr",
    stdin='[{"size": 100, "stencil": "7pt"}]',
)
