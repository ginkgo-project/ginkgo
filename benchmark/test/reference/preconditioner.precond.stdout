[
    {
        "size": 100,
        "stencil": "7pt",
        "preconditioner": {
            "jacobi-32-0,0": {
                "generate": {
                    "components": {
                        "generate(<typename>)": 1.0,
                        "allocate": 1.0,
                        "jacobi::find_blocks": 1.0,
                        "jacobi::generate": 1.0,
                        "free": 1.0,
                        "overhead": 1.0
                    },
                    "time": 1.0,
                    "repetitions": 10
                },
                "apply": {
                    "components": {
                        "apply(<typename>)": 1.0,
                        "jacobi::simple_apply": 1.0,
                        "overhead": 1.0
                    },
                    "time": 1.0,
                    "repetitions": 10
                },
                "completed": true
            }
        },
        "rows": 125,
        "cols": 125,
        "nonzeros": 725
    }
]
